app/src/main/java/org/gnucash/android/ui/settings/BackupPreferenceFragment.java

```diff
@@ -292,64 +296,75 @@ private void toggleOwnCloudSync(Preference pref){
 		}
 	}
 
-
 	public static GoogleApiClient getGoogleApiClient(final Context context) {
-		return new GoogleApiClient.Builder(context)
+		return getGoogleApiClient(context, null);
+	}
+
+	public static GoogleApiClient getGoogleApiClient(final Context context, OnCloudStorageListener listener) {
+		final WeakReference<OnCloudStorageListener> weakListener = new WeakReference<>(listener);
```

Comment:
I'm not sure I see the need for the `WeakReference`. `listener` is only set to a non-null value when called internally from `onCreate()`. So if the Fragment is destroyed, the `GoogleApiClient` should be garbage collected, shouldn't it?